basyx/aas/model/base.py

```diff
@@ -1734,15 +1716,45 @@ def __init__(self) -> None:
         super().__init__()
         self.namespace_element_sets: List[NamespaceSet] = []
 
-    def get_referable(self, id_short: NameType) -> Referable:
+    def get_referable(self, id_short: Union[NameType, Iterable[NameType]]) -> Referable:
         """
-        Find a :class:`~.Referable` in this Namespace by its id_short
+        Find a :class:`~.Referable` in this Namespace by its id_short or by its id_short path.
+        The id_short path may contain :class:`~basyx.aas.model.submodel.SubmodelElementList` indices.
 
-        :param id_short: id_short
+        :param id_short: id_short or id_short path as any :class:`Iterable`
         :returns: :class:`~.Referable`
+        :raises TypeError: If one of the intermediate objects on the path is not a
+                           :class:`~.UniqueIdShortNamespace`
+        :raises ValueError: If a non-numeric index is given to resolve in a
+                            :class:`~basyx.aas.model.submodel.SubmodelElementList`
         :raises KeyError: If no such :class:`~.Referable` can be found
         """
-        return super()._get_object(Referable, "id_short", id_short)  # type: ignore
+        from .submodel import SubmodelElementList
+        if isinstance(id_short, NameType):
```

Comment:
One thought: As far as I could just see when quickly checking, we do not make sure in `Referable._set_id_short()` that a user actually supplied a `NameType`, do we?
In case a user just simply sets it as a string that conforms to `NameType` constraints, we simply hand over the string to `self._id_short`. 
This would break the resolving here, wouldn't it?

Reply:
> we do not make sure in `Referable._set_id_short()` that a user actually supplied a `NameType`, do we?

We do, this is checked in `Referable.validate_id_short()`, which is called in `Referable._set_id_short()`.